### DISCUSSION TITLE
CSV/Excel File Import

### DISCUSSION THREAD

**Discussion by AndreyKindin** [directus/directus#7246] [Feature Requests]
It would be nice if besides exporting CSV there was also the possibility to import the CSV directly in the Directus app.
Thank you and greetings Andrey

**Comment by rijkvanzanten** [comment]
Hi Andrey!


Yes u can already do that through https://docs.directus.io/reference/api/system/utilities/#import-data-from-file 👍🏻☺️

**Reply by AndreyKindin** [reply]
Hi Rijk,
I know about the API possibility and that's great
But not what I mean I'm talking about the DIrectus GUI. A button to upload the file/select from the file library, and where I can select the respective columns to import. Plus it would be great if the standard columns (id,status,user_cr...,date_cre..) were automatically filled.
I know it's a lot but it's an idea and it would be great.

**Reply by rijkvanzanten** [reply]
Ahh gotcha 👍🏻

**Reply by adojo-1** [reply]
We are also needing the ability to upload a CSV file to easily import data in mass into a directus collection without having to do custom api programming, but instead just select a csv file to import the data through the directus admin gui.

**Reply by rijkvanzanten** [reply]
@adojo-1 Just out of curiosity, when you say _to easily import data in mass_, what sort of data-set size are we talking? Couple hundred/thousand/more rows?

**Comment by adojo-1** [comment]
Good question... There will often be spreadsheet CSV files we import that
have around 5000 to 10,000 rows of data.

There were be some files from time to time that have around 50,000 to
75,000 rows

Dave

**Reply by rijkvanzanten** [reply]
Thanks! We'll have to do some research into how we can efficiently import those, without blowing up the CPU of your server 😄 We might have to stream the file to the local file system first, and then process the file in chunks of rows, or potentially make it admin only so we can skip the permissions checks for each row 🤔 Lots to think through!

**Comment by rijkvanzanten** [comment]
This has since been implemented!
